pkg/apis/v1/nodepool_validation_cel_test.go

```diff
@@ -697,4 +700,148 @@ var _ = Describe("CEL/Validation", func() {
+				nodePool.Spec.Weight = lo.ToPtr(int32(25))
+				nodePool.Spec.Limits = Limits{
+					v1.ResourceCPU:    resource.MustParse("1000"),
+					v1.ResourceMemory: resource.MustParse("2000Gi"),
+					"nodes":           resource.MustParse("100"),
+				}
+				nodePool.Spec.Disruption = Disruption{
+					ConsolidationPolicy: ConsolidationPolicyWhenEmptyOrUnderutilized,
+					ConsolidateAfter:    MustParseNillableDuration("5m"),
+					Budgets: []Budget{
+						{
+							Nodes:    "20%",
+							Schedule: lo.ToPtr("0 9 * * mon-fri"),
+							Duration: &metav1.Duration{Duration: 8 * time.Hour},
+							Reasons:  []DisruptionReason{DisruptionReasonDrifted, DisruptionReasonEmpty},
+						},
+						{
+							Nodes:    "30%",
+							Schedule: lo.ToPtr("0 22 * * sat,sun"),
+							Duration: &metav1.Duration{Duration: 4 * time.Hour},
+							Reasons:  []DisruptionReason{DisruptionReasonUnderutilized},
+						},
+						{
+							Nodes: "5",
+						},
+					},
+				}
+				nodePool.Spec.Template.Labels = map[string]string{
+					"environment": "production",
+				}
+				nodePool.Spec.Template.Annotations = map[string]string{
+					"example.com/last-updated": "2024-01-01",
+				}
+				nodePool.Spec.Template.Spec.Taints = []v1.Taint{
+					{Key: "gpu-workload", Value: "enabled", Effect: v1.TaintEffectNoSchedule},
+				}
+				nodePool.Spec.Template.Spec.StartupTaints = []v1.Taint{
+					{Key: "example.com/initializing", Value: "true", Effect: v1.TaintEffectNoSchedule},
+				}
+				nodePool.Spec.Template.Spec.Requirements = []NodeSelectorRequirementWithMinValues{
+					{NodeSelectorRequirement: v1.NodeSelectorRequirement{Key: v1.LabelInstanceTypeStable, Operator: v1.NodeSelectorOpIn, Values: []string{"m5.large", "m5.xlarge", "m5.2xlarge", "c5.large", "c5.xlarge", "r5.large", "r5.xlarge"}}, MinValues: lo.ToPtr(3)},
+					{NodeSelectorRequirement: v1.NodeSelectorRequirement{Key: v1.LabelTopologyZone, Operator: v1.NodeSelectorOpIn, Values: []string{"us-west-2a", "us-west-2b", "us-west-2c"}}},
+					{NodeSelectorRequirement: v1.NodeSelectorRequirement{Key: CapacityTypeLabelKey, Operator: v1.NodeSelectorOpIn, Values: []string{"on-demand", "spot"}}},
+					{NodeSelectorRequirement: v1.NodeSelectorRequirement{Key: v1.LabelArchStable, Operator: v1.NodeSelectorOpIn, Values: []string{"amd64"}}},
+					{NodeSelectorRequirement: v1.NodeSelectorRequirement{Key: "karpenter.k8s.aws/instance-memory", Operator: v1.NodeSelectorOpGt, Values: []string{"8192"}}},
+					{NodeSelectorRequirement: v1.NodeSelectorRequirement{Key: "karpenter.k8s.aws/instance-cpu", Operator: v1.NodeSelectorOpLt, Values: []string{"64"}}},
+				}
+
+				nodePool.Spec.Template.Spec.ExpireAfter = MustParseNillableDuration("72h")
+				nodePool.Spec.Template.Spec.TerminationGracePeriod = &metav1.Duration{Duration: time.Second * 300}
+
+				Expect(env.Client.Create(ctx, nodePool)).To(Succeed())
+				Expect(nodePool.RuntimeValidate(ctx)).To(Succeed())
+			})
+		})
+
+		Context("Update Scenarios", func() {
```

Comment:
Should we also have a test for updating replicas from one valid value to another?